test/e2e/federation-util.go

```diff
@@ -300,14 +300,18 @@ func cleanupServiceShardsAndProviderResources(namespace string, service *v1.Serv
 				// Get failed with an error, try again.
 				framework.Logf("Failed to find service %q in namespace %q, in cluster %q: %v. Trying again in %s", service.Name, namespace, name, err, framework.Poll)
 				return false, nil
+			} else if errors.IsNotFound(err) {
+				cSvc = nil
+				By(fmt.Sprintf("Service %q in namespace %q in cluster %q not found", service.Name, namespace, name))
+				return true, nil
```

Comment:
Good idea. Returned the error. There is nothing special to do about it outside anyway. If it is not found, we just move on, as is the case with any other error. Because, we have waited for long enough already.